File: readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <math.h>
#include <ctype.h>

int main(void)
{
    // getting user input
    string text = get_string("Text: ");

    int i = 0;
    int letters = 0;
    int words = 1;
    int sentences = 0;

    // looping through each character of string
    while (i < strlen(text))
    {
        // test for if letter
        if ((text[i] >= 'a' && text[i] <= 'z') || (text[i] >= 'A' && text[i] <= 'Z'))
        {
            letters++;
        }

        // test for word end
        else if (text[i] == ' ')
        {
            words++;
        }

        else if (i > 0 && (text[i] == '!' || text[i] == '?' || text[i] == '.') && isalnum(text[i - 1]))
        {

            sentences++;
        }
        i++;
    }

    // main algorithm
    float grade = 0.0588 * (100 * (float) letters / (float) words) - 0.296 * (100 * (float) sentences / (float) words) - 15.8;


    if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", (int) round(grade));
    }
}

File: readability.py
# code for number of sentences from https://stackoverflow.com/questions/3549075/regex-to-find-all-sentences-of-text
from cs50 import get_string
import re

# getting user input and setting counter
text = get_string('Text: ')
alphabets = 0

# looping through each char to find number of alphabets
for i in range(len(text)):
    if text[i].isalpha():
        alphabets += 1

# creating a list by spliting and finding its lenght
word_list = text.split(" ")
words = len(word_list)

# finding number of sentences using regex
pat = re.compile(r'([A-Z][^\.!?]*[\.!?])', re.M)
sentences = len(pat.findall(text))

# main algorthim
L = (alphabets/words) * 100
S = (sentences/words) * 100
grade = 0.0588 * L - 0.296 * S - 15.8

# output
if grade < 1:
    print("Before Grade 1")
elif grade > 16:
    print("Grade 16+")
else:
    print(f"Grade {round(grade)}")